api/handlers.go

```diff
@@ -15,8 +18,17 @@ type Sessions interface {
 	New(email string) (*session.Session, error)
 }
 
+// Cache interface for storing sessions
+type Cache interface {
+	Set(s *session.Session)
```

Comment:
It might be worthwhile adding an error return type to this method - I think it's a safe bet that the real implementation could run into issues.